### PR TITLE
Chore/581

### PR DESCRIPTION
**Pull request by vipergtsrz**
- Include token swap status component in the deal dashboard page.
- Move execute swap functionality from funding to the deal dashboard page.
- Create a <status-action-bar> component for reusability under the token status component
- Convert the ITokenFunding interface to ITokenCalculated and add properties to allow one interface to work with funding tokens as well as claiming tokens
- Add an isFullyClaimed getter to DealTokenSwap to check if a deal's tokens have been fully claimed or not
- Rewrite some logic for the PL view to handle the PL also being a representative of a DAO
- Rewrite the logic of the status card component to handle claiming tokens as well as funding tokens
- Update the swap status component to work how the SOT changed it to
- Fix a bug in the pselect component that didn't allow the value to be preselected

### CODE REVIEW

File: src/dealDashboard/dealDashboard.scss [hunk 1]
```diff
@@ -1,6 +1,51 @@
 @import "src/styles/styles.scss";
 @import "src/resources/elements/primeDesignSystem/styles";
-
+.ppopup-dialog.executeSwap {
```
Comment (dkent600) [inline src/dealDashboard/dealDashboard.scss:3]: this should be in its own modal dialog component

File: src/dealDashboard/dealDashboard.ts [hunk 2]
```diff
@@ -51,6 +56,10 @@ export class DealDashboard {
         this.eventAggregator.publish("showMessage", "Deal privacy has been successfully submitted");
       }),
     );
+    //wait until the dao transactions from the contract are there
+    await Utils.waitUntilTrue(() => this.deal.daoTokenTransactions !== undefined);
```
Comment (dkent600) [inline src/dealDashboard/dealDashboard.ts:60]: both of these need to be located in the components that require this data

File: src/dealDashboard/dealDashboard.ts [hunk 3]
```diff
@@ -60,4 +69,73 @@ export class DealDashboard {
...
+      this.isExecutingSwap = false;
+    }
+  }
+
+  /**
+   * Gets the HTML for the dao token swap modal popup
+   * @param dao
+   * @returns string
+   */
+  private getDaoHtmlForSwap(dao: IDAO): string {
```
Comment (dkent600) [inline src/dealDashboard/dealDashboard.ts:118]: can we avoid this definition of HTML in the viewmodel?

File: src/dealDashboard/dealDashboard.ts [hunk 4]
```diff
@@ -60,4 +69,73 @@ export class DealDashboard {
...
+      </div>
+      `;
+  }
+
+  /**
+   * Opens a new window to the transaction id or address on the blockchain
+   * @param address
+   * @param tx
+   */
+  private gotoEtherscan = (address: string, tx = false): void => {
```
Comment (dkent600) [inline src/dealDashboard/dealDashboard.ts:138]: use normal typescript function definition syntax

File: src/resources/elements/address-link/address-link.html [hunk 5]
```diff
@@ -1,6 +1,10 @@
 <template>
   <div class="addressLink">
-    <a if.to-view="textClickable" ptooltip.to-view="address" click.delegate="gotoEtherscan()">
+    <a
+      if.to-view="textClickable"
+      ptooltip.to-view="address && showTooltip ? address : null"
+      click.delegate="gotoEtherscan()"
```
Comment (dkent600) [inline src/resources/elements/address-link/address-link.html:6]: why aren't you using the ethercanlink component?

File: src/resources/elements/primeDesignSystem/pselect/pselect.ts [hunk 6]
```diff
@@ -44,12 +45,16 @@ export class PSelect {
...
   @bindable({defaultBindingMode: bindingMode.twoWay}) value?: string | string[];
   @bindable validationState?: ValidationState;
+  constructor(private readonly taskQueue: TaskQueue){}
 
   refSelectInput: HTMLSelectElement;
   select: SlimSelect;
   isOpen = false;
 
   attached(): void {
+    this.taskQueue.queueTask(() => this.select?.set(this.value));
```
Comment (dkent600) [inline src/resources/elements/primeDesignSystem/pselect/pselect.ts:55]: i didn't know about taskqueue.  read about it once and forget.  wish I had remembered.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/dealDashboard/dealDashboard.html [hunk 7]
```diff
@@ -39,13 +45,55 @@ <h2 data-test="dealStatus">
       authorized.bind="isAllowedToDiscuss"
       deal.bind="deal"
       discussion-id.two-way="discussionId"
-      if.to-view="deal.isUserRepresentativeOrLead || !deal.isPrivate"></deal-clauses>
-
+      if.to-view="deal.isUserRepresentativeOrLead || !deal.isPrivate"
+    ></deal-clauses>
+    <swap-status deal.bind="deal" if.to-view="deal.primaryDao.tokens && deal.partnerDao">
```
[the remark this answers is not in the document]
Comment (vipergtsrz) [reply src/dealDashboard/dealDashboard.html]: Done

File: src/dealDashboard/dealDashboard.scss (hunk 1 above)
Comment (vipergtsrz) [reply src/dealDashboard/dealDashboard.scss]: Done

File: src/dealDashboard/dealDashboard.ts (hunk 2 above)
Comment (vipergtsrz) [reply src/dealDashboard/dealDashboard.ts]: Done

File: src/dealDashboard/dealDashboard.ts (hunk 3 above)
Comment (vipergtsrz) [reply src/dealDashboard/dealDashboard.ts]: This HTML was moved to the modal component as requested

File: src/dealDashboard/dealDashboard.ts (hunk 4 above)
Comment (vipergtsrz) [reply src/dealDashboard/dealDashboard.ts]: This method has been removed

File: src/resources/elements/address-link/address-link.html (hunk 5 above)
Comment (vipergtsrz) [reply src/resources/elements/address-link/address-link.html:6]: The <etherscanlink> component doesn't have any pop out icon in it so the address link component handles that
image

**Review by dkent600 — APPROVED** [review]

**Pull request merged**